lxd/storage/drivers/clients/alletra_wsapi1.go

```diff
@@ -837,3 +845,49 @@ func (p *AlletraClient) DeleteVolumeSnapshot(poolName string, volName string, sn
 
 	return nil
 }
+
+// getTaskState retrieves a running task state.
+func (p *AlletraClient) getTaskState(taskID string) (*hpeTaskState, error) {
+	var resp hpeTaskState
+
+	url := api.NewURL().Path("api", "v1", "tasks", taskID).WithQuery("view", "excludeDetail")
+	err := p.requestAuthenticated(http.MethodGet, url.URL, nil, &resp)
+	if err != nil {
+		if isHpeErrorNotFound(err) {
+			return nil, api.StatusErrorf(http.StatusNotFound, "Task with ID %q not found", taskID)
+		}
+
+		return nil, fmt.Errorf("Failed to retrieve task with ID %q: %w", taskID, err)
+	}
+
+	return &resp, nil
+}
+
+// waitTaskFinish waits for HPE Alletra task to finish with any result (error, canceled, success).
+func (p *AlletraClient) waitTaskFinish(ctx context.Context, taskID string) (int, error) {
+	_, ok := ctx.Deadline()
+	if !ok {
+		// Set a default timeout of 180 seconds for the context
+		// if no deadline is already configured.
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, 180*time.Second)
+		defer cancel()
+	}
+
+	for {
+		taskState, err := p.getTaskState(taskID)
+		if err != nil {
+			return -1, err
+		}
+
+		if taskState.Status != taskStatusActive {
+			return taskState.Status, nil
+		}
+
+		if ctx.Err() != nil {
```

Comment:
tiny nit, but you're calling Err() twice here, if it returns non-nil, so suggest assigning it to `err` first and then comparing/returning it.